scripts/prepare_sector_network.py

```diff
@@ -874,9 +872,28 @@ def add_generation(n, costs):
             * costs.at[generator, "VOM"],  # NB: VOM is per MWel
             capital_cost=costs.at[generator, "efficiency"]
             * costs.at[generator, "fixed"],  # NB: fixed cost is per MWel
-            p_nom_extendable=True,
+            p_nom_extendable=(
+                True
+                if generator
+                in snakemake.params.electricity.get("extendable_carriers", dict()).get(
+                    "Generator", list()
+                )
+                else False
+            ),    
+            p_nom=(
+                existing_capacities[generator] / costs.at[generator, "efficiency"]
```

Comment:
Here it needs to be `existing_capacities[generator] / existing_efficiencies[generator]`. As I have suggested previosuly in the chat.

Reply:
If you want you can directly make a code suggestion, makes it easier to apply your changes